🐛 GraphLogger NPE when checking status of runs (possible race condition)

Only occurred once, but its possible that it is a race condition when polling the rdpc gateway before the workflow shows up. 

Stacktrace:
```
Caused by: java.lang.NullPointerException
	at org.icgc_argo.workflowgraphnode.logging.GraphLogger.graphLog(GraphLogger.java:139) ~[classes!/:?]
	at org.icgc_argo.workflowgraphnode.logging.GraphLogger.info(GraphLogger.java:24) ~[classes!/:?]
	at org.icgc_argo.workflowgraphnode.rabbitmq.NodeConfiguration.lambda$runningToComplete$0(NodeConfiguration.java:101) ~[classes!/:?]
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:189) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxContextStart$ContextStartSubscriber.onNext(FluxContextStart.java:103) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxConcatArray$ConcatArraySubscriber.onNext(FluxConcatArray.java:176) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:121) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxHandleFuseable$HandleFuseableSubscriber.tryOnNext(FluxHandleFuseable.java:119) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxDoFinally$DoFinallyFuseableConditionalSubscriber.tryOnNext(FluxDoFinally.java:248) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.tryOnNext(FluxPeekFuseable.java:522) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.tryOnNext(FluxPeekFuseable.java:522) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxDoFinally$DoFinallyFuseableConditionalSubscriber.tryOnNext(FluxDoFinally.java:248) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableConditionalSubscriber.tryOnNext(FluxPeekFuseable.java:522) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxOnAssembly$OnAssemblyConditionalSubscriber.tryOnNext(FluxOnAssembly.java:529) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxPublishOn$PublishOnConditionalSubscriber.runAsync(FluxPublishOn.java:866) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxPublishOn$PublishOnConditionalSubscriber.run(FluxPublishOn.java:939) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.scheduler.ImmediateScheduler$ImmediateSchedulerWorker.schedule(ImmediateScheduler.java:79) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxPublishOn$PublishOnConditionalSubscriber.trySchedule(FluxPublishOn.java:759) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxPublishOn$PublishOnConditionalSubscriber.onNext(FluxPublishOn.java:704) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onNext(MonoFlatMapMany.java:242) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onNext(FluxDoFinally.java:123) ~[reactor-core-3.3.2.RELEASE.jar!/:3.3.2.RELEASE]
	at reactor.rabbitmq.Sender$PublishConfirmSubscriber$1.handleAckNack(Sender.java:953) [reactor-rabbitmq-1.4.1.RELEASE.jar!/:1.4.1.RELEASE]
	at reactor.rabbitmq.Sender$PublishConfirmSubscriber$1.handleAck(Sender.java:928) [reactor-rabbitmq-1.4.1.RELEASE.jar!/:1.4.1.RELEASE]
	at com.rabbitmq.client.impl.ChannelN.callConfirmListeners(ChannelN.java:498) [amqp-client-5.7.3.jar!/:5.7.3]
	at com.rabbitmq.client.impl.ChannelN.processAsync(ChannelN.java:369) [amqp-client-5.7.3.jar!/:5.7.3]
	at com.rabbitmq.client.impl.AMQChannel.handleCompleteInboundCommand(AMQChannel.java:182) [amqp-client-5.7.3.jar!/:5.7.3]
	at com.rabbitmq.client.impl.AMQChannel.handleFrame(AMQChannel.java:114) [amqp-client-5.7.3.jar!/:5.7.3]
	at com.rabbitmq.client.impl.AMQConnection.readFrame(AMQConnection.java:672) [amqp-client-5.7.3.jar!/:5.7.3]
	at com.rabbitmq.client.impl.AMQConnection.handleReadFrame(AMQConnection.java:621) [amqp-client-5.7.3.jar!/:5.7.3]
	at com.rabbitmq.client.impl.nio.NioLoop.run(NioLoop.java:160) [amqp-client-5.7.3.jar!/:5.7.3]
```

--- a/src/main/java/org/icgc_argo/workflowgraphnode/components/Workflows.java
+++ b/src/main/java/org/icgc_argo/workflowgraphnode/components/Workflows.java
@@ -1,7 +1,5 @@
 package org.icgc_argo.workflowgraphnode.components;
 
-import static org.icgc_argo.workflowgraphnode.service.GraphTransitAuthority.*;
-
 import com.pivotal.rabbitmq.stream.Transaction;
 import java.util.List;
 import java.util.UUID;
@@ -9,7 +7,9 @@ import java.util.function.BiConsumer;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import lombok.val;
+import org.icgc_argo.workflow_graph_lib.exceptions.CommittableException;
 import org.icgc_argo.workflow_graph_lib.exceptions.DeadLetterQueueableException;
+import org.icgc_argo.workflow_graph_lib.exceptions.RequeueableException;
 import org.icgc_argo.workflow_graph_lib.schema.GraphEvent;
 import org.icgc_argo.workflow_graph_lib.schema.GraphRun;
 import org.icgc_argo.workflow_graph_lib.workflow.client.RdpcClient;
@@ -53,16 +53,16 @@ public class Workflows {
           s -> {
             if (ROLLBACK.contains(s)) {
               GraphLogger.debug(tx, "Requeueing %s with status %s", tx.get().getRunId(), s);
-              requeueAndRemoveTransactionFromGTA(tx);
+              sink.error(new RequeueableException(tx.get().getRunId()));
             } else if (NEXT.contains(s)) {
               GraphLogger.debug(tx, "Nexting %s with status %s", tx.get().getRunId(), s);
               sink.next(tx);
             } else if (REJECT.contains(s)) {
               GraphLogger.debug(tx, "Rejecting %s with status %s", tx.get().getRunId(), s);
-              rejectAndRemoveTransactionFromGTA(tx);
+              sink.error(new DeadLetterQueueableException(tx.get().getRunId()));
             } else if (COMMIT.contains(s)) {
               GraphLogger.debug(tx, "Committing %s with status %s", tx.get().getRunId(), s);
-              commitAndRemoveTransactionFromGTA(tx);
+              sink.error(new CommittableException(tx.get().getRunId()));
             } else {
               GraphLogger.error(tx, "Cannot map workflow status for run: %s.", tx.get().getRunId());
               sink.error(new DeadLetterQueueableException(tx.get().getRunId()));

--- a/src/main/java/org/icgc_argo/workflowgraphnode/service/GraphTransitAuthority.java
+++ b/src/main/java/org/icgc_argo/workflowgraphnode/service/GraphTransitAuthority.java
@@ -87,7 +87,8 @@ public class GraphTransitAuthority {
 
   /**
    * Commits the transaction and removes it's associated GraphTransitObject from the
-   * GraphTransitAuthority registry
+   * GraphTransitAuthority registry (should only be used in the Errors component or as the final
+   * subscribe consumer)
    *
    * @param tx the transaction to be committed and for which the corresponding GTO should be removed
    *     from the GTA
@@ -99,7 +100,7 @@ public class GraphTransitAuthority {
 
   /**
    * Rejects the transaction and removes it's associated GraphTransitObject from the
-   * GraphTransitAuthority registry
+   * GraphTransitAuthority registry (should only be used in the Errors component)
    *
    * @param tx the transaction to be rejected and for which the corresponding GTO should be removed
    *     from the GTA
@@ -111,7 +112,7 @@ public class GraphTransitAuthority {
 
   /**
    * Requeue (tx.rollback(true)) the transaction and removes it's associated GraphTransitObject from
-   * the GraphTransitAuthority registry
+   * the GraphTransitAuthority registry (should only be used in the Errors component)
    *
    * @param tx the transaction that will requeue and for which the corresponding GTO should be
    *     removed from the GTA

--- a/src/test/java/org/icgc_argo/workflowgraphnode/components/WorkflowsTest.java
+++ b/src/test/java/org/icgc_argo/workflowgraphnode/components/WorkflowsTest.java
@@ -89,12 +89,11 @@ public class WorkflowsTest {
                 })
             .collect(toList());
 
-    val handler = Workflows.handleRunStatus(rdpcClientMock);
-
     val source =
         Flux.fromIterable(runIdTransactions)
             .doOnNext(graphTransitAuthority::registerGraphRunTx)
-            .handle(handler);
+            .handle(Workflows.handleRunStatus(rdpcClientMock))
+            .onErrorContinue(Errors.handle());
 
     StepVerifier.create(source)
         // transaction 0 is sent to the next call unchanged in the flux handler

